addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/UrlUtils.java

```diff
@@ -0,0 +1,162 @@
+    private static HttpSender sender =
+            new HttpSender(
+                    Model.getSingleton().getOptionsParam().getConnectionParam(),
+                    true,
+                    HttpSender.MANUAL_REQUEST_INITIATOR);
+    // private static HttpRequestConfig config =
+    // HttpRequestConfig.builder().setSoTimeout(10).build();
+
+    /**
+     * Returns a List of all parameters to be used from a given wordlist file.
+     *
+     * @param path the path to the wordlist.
+     * @return a List<String> of all parameters to be used from a given wordlist file.
+     */
+    public static List<String> reader(String path) {
+        logger.info("Reading keys from file: " + path);
+        List<String> params = new ArrayList<>();
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(path));
+            String line;
+            while ((line = br.readLine()) != null) {
+                params.add(StringUtils.strip(line));
+            }
+            br.close();
+            logger.info("Succesfully parsed keys!");
+        } catch (Exception e) {
+            logger.error(e);
+        }
+        return params;
+    }
+
+    /**
+     * Returns "crashproof" HTTP Responses from a given URL by using GET method.
+     *
+     * @param url the url to be tested.
+     * @param headers the associated headers.
+     * @return a HttpMessage of the HTTP Response from the given URL.
+     */
+    public static HttpMessage stableRequest(String url, HttpRequestHeader headers) {
+        try {
+            URI parsed = new URI(url, true);
+            String scheme = parsed.getScheme();
+            String host = parsed.getHost();
+            String path = parsed.getPath();
+            String port = parsed.getPort() == -1 ? "" : Integer.toString(parsed.getPort());
+            String[] schemes =
+                    (scheme.equalsIgnoreCase("https"))
+                            ? new String[] {"https", "http"}
+                            : new String[] {"http", "https"};
+            int errorCount = 0;
+            for (String s : schemes) {
+                try {
+                    HttpMessage msg = new HttpMessage();
+                    headers.setURI(new URI(s + "://" + host + ":" + port + path, true));
+                    headers.setMethod("GET");
+                    msg.setRequestHeader(headers);
+                    sender.sendAndReceive(msg);
+                    return msg;
+                } catch (Exception e) {
+                    // Check if it is a connection error
+                    if (e.getMessage().contains("ConnectionError")
+                            || e.getMessage().contains("IOException")) {
+                        continue;
+                    }
+                    logger.info(e);
+                    errorCount = errorCount + 1;
+                }
+                logger.info("Total number of Errors [stableRequest()]: " + errorCount);
```

Comment:
- Should be logged just once, after the loop.
 - The method name is unnecessary.

Reply:
Uhh what can be the name then?

Reply:
>The method name is unnecessary.
What should be done about this?

Reply:
Just remove it, the name is not needed, I assume this is for debug, if it's for the user it should not be (just) in the logs but in the UI or the data returned by the API.

Reply:
Oh in the debug statement !!!! :) My IQ is dropping day by day :) I thought the entire function name and I'm wondering how should I call a function without a function name ... Does he want me to rename it :). sorry :)